kube/secret.go

```diff
@@ -50,9 +60,27 @@ func MakeSecrets(secrets model.CVMap, defaults map[string]string) ([]*apiv1.Secr
 	total := 0 // Accumulated size of the values stored in 'currentSecret'
 
 	for key, value := range secrets {
-		ok, strValue := configValue(value, defaults)
-		if !ok {
-			return nil, nil, fmt.Errorf("Secret '%s' has no value", key)
+		var strValue string
+		if createHelmChart {
+			switch {
+			case value.Generator == nil || value.Generator.Type != model.GeneratorTypePassword:
```

Comment:
I don't see how you can do this with `default`.  You need to cover the `value.Generator == nil` case first before you can handle `case value.Generator.Type == model.GeneratorTypePassword`. So you would then end up with duplicate code.

Lmk if I'm not seeing the obvious here!